src/xenia/gpu/graphics_system.cc

```diff
@@ -118,6 +118,20 @@ X_STATUS GraphicsSystem::Setup(cpu::Processor* processor,
       reinterpret_cast<cpu::MMIOReadCallback>(ReadRegisterThunk),
       reinterpret_cast<cpu::MMIOWriteCallback>(WriteRegisterThunk));
 
+  // GPU counters update thread
+  auto counters_worker_thread = kernel::object_ref<kernel::XHostThread>(
+      new kernel::XHostThread(kernel_state_, 128 * 1024, 0, [this]() {
+        while (1) {
```

Comment:
This is going to be the most CPU-intensive thread in the entire emulator likely, plus the units of the counter are completely random now, just tied to the host CPU/memory/OS performance.

Reply:
I agree with it. That is why I marked it as "SUBJECT TO CHANGE" and wrote that probably there should be some conditions when/how often it is iterated 💃